neuro_san/test/evaluators/gist_agent_evaluator.py

```diff
@@ -26,6 +27,27 @@ class GistAgentEvaluator(AbstractAgentEvaluator):
     match the gist of the value to be verified against.
     """
 
+    SHIPPED_HOCONS = FileOfClass(__file__, path_to_basis="../../registries")
+
+    def __init__(self, asserts: AssertForwarder, negate: bool = False,
+                 discriminator_agent: str = None, connection_type: str = "direct"):
+        """
+        Constructor
+
+        :param asserts: The AssertForwarder instance to handle failures
+        :param negate: If true the verification criteria should be negated
+        :param discriminator_agent: The name of the discriminator agent to use for the test
+                                By default this is None, implying the stock "gist.hocon"
+                                agent shipped with neuro-san should be used.
+        :param connection_type: The string connection type to pass to the AgentSessionFactory
+        """
+        super().__init__(asserts, negate)
+        self.connection_type: str = connection_type
+        self.discriminator_agent: str = discriminator_agent
+
+        if discriminator_agent is None:
+            self.discriminator_agent = self.SHIPPED_HOCONS.get_file_in_basis("gist.hocon")
```

Comment:
Allowo the GistAgentEvaluator to take different connection types and/or different agent references.
The default is the "gist.hocon" that ships with the neuro-san toolkit.